src/Internal/Service/Error.purs

```diff
@@ -0,0 +1,89 @@
+module Ctl.Internal.Service.Error
+  ( BlockfrostError(BlockfrostError)
+  , ClientError
+      ( ClientHttpError
+      , ClientHttpResponseError
+      , ClientDecodeJsonError
+      , ClientEncodingError
+      , ClientOtherError
+      )
+  , ServiceError
+      ( ServiceBlockfrostError
+      , ServiceOtherError
+      )
+  ) where
+
+import Prelude
+
+import Aeson (class DecodeAeson, JsonDecodeError, getField)
+import Affjax (Error, printError) as Affjax
+import Ctl.Internal.Service.Helpers (aesonObject)
+import Data.Generic.Rep (class Generic)
+import Data.Newtype (class Newtype)
+import Data.Show.Generic (genericShow)
+
+--------------------------------------------------------------------------------
+-- ClientError
+--------------------------------------------------------------------------------
+
+data ClientError
+  = ClientHttpError Affjax.Error
+  | ClientHttpResponseError ServiceError
```

Comment:
Can we add the status code to `ClientHttpResponseError`, like I have in my PR? https://github.com/Plutonomicon/cardano-transaction-lib/blob/a4b545e29e9e8f9a37823f1f37699f72660a6158/src/Internal/QueryM.purs#L390

I needed it to check for 404 errors to not have them be a `ClientError`, but a richer error or the result type (`isTxConfirmed` for eg will check for 404 at the `/txs/{hash}` endpoint and return `false`)

It's likely other endpoints may need to check for 404 too